Toolkit/Public/Get-RscNasSystem.ps1

```diff
@@ -23,33 +26,57 @@ function Get-RscNasSystem {
 
     .EXAMPLE
     Get details of NAS system with id="72859a28-6276-555a-9a66-d93fe99d2751"
+    Get-RscNasSystem -Id "72859a28-6276-555a-9a66-d93fe99d2751"
+        Or 
     Get-RscNasSystem "72859a28-6276-555a-9a66-d93fe99d2751"
 
     .EXAMPLE
-    Retrieve list of NAS systems.
+    Retrieve list of NAS systems (first 2).
     Get-RscNasSystem -First 2
+    
+    .EXAMPLE
+    Retrieve all NAS systems. 
+    It will return at max 50 items (default page size).
+    Get-RscNasSystem
+
+    .EXAMPLE
+    Retrieve all NAS systems with the name containing "foo".
+    It will return at max 50 items (default page size).
+    Get-RscNasSystem -Name "foo"
+
+    .EXAMPLE
+    Retrieve first 4 NAS systems with the name containing "foo".
+    Get-RscNasSystem -Name "foo" -First 4
     #>
 
     [CmdletBinding(
-        DefaultParameterSetName = "Id"
+        DefaultParameterSetName = "NameAndList"
     )]
     Param(
         # The UUID of the object representing the NAS system.
         [Parameter(
             ParameterSetName = "Id",
-            Mandatory = $true,
+            Mandatory = $false,
             Position = 0
         )]
         [ValidateNotNullOrEmpty()]
         [String]$Id,
 
+        # The name of the NAS system to filter on.
+        [Parameter(
+            ParameterSetName = "NameAndList",
+            Mandatory = $false
+        )]
+        [ValidateNotNullOrEmpty()]
+        [String]$Name,
+        
         # The number of NAS systems to retrieve in one query.
         [Parameter(
-            ParameterSetName = "List",
+            ParameterSetName = "NameAndList",
             Mandatory = $false
         )]
         [Int]$First = 50,
```

Comment:
we can probably just remove this while we're at it.

Reply:
Removed '$First' input parameter. Now have 'List' switch parameter, which is also a Default parameter.